### PR TITLE
[flutter_tools] Concise dart2js commands outputKernel & outputJavaScript

### PR DESCRIPTION
**Pull request by DetectiveAI4869**
## Description

Concise dart2js commands outputKernel & outputJavaScript. 

Remove once variables
Extract public parameters

## Related Issues

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR. There should be at least one issue listed here.*

## Tests

I added the following tests:

Reuse Dart2JSTarget tests

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by includecmath** [comment]
@jonahwilliams

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 1]
```diff
@@ -153,64 +153,53 @@ class Dart2JSTarget extends Target {
...
-    final String specPath = globals.fs.path.join(
-      globals.artifacts.getArtifactPath(Artifact.flutterWebSdk), 'libraries.json');
-    // TODO(jonahwilliams): support configuration of this file.
-    const String packageFile = '.packages';
-    final File outputKernel = environment.buildDir.childFile('app.dill');
-    final File outputFile = environment.buildDir.childFile('main.dart.js');
-    final List<String> dartDefines = decodeDartDefines(environment.defines, kDartDefines);
-    final List<String> extraFrontEndOptions = decodeDartDefines(environment.defines, kExtraFrontEndOptions);
+
+    List<String> initCommand() {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart:158]: `initCommand` doesn't need to be a function, you can re-use the same List<String> in both spreads.

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 2]
```diff
@@ -153,64 +153,53 @@ class Dart2JSTarget extends Target {
...
+          '-Ddart.vm.profile=true'
+        else
+          '-Ddart.vm.product=true',
+        for (final String dartDefine
+            in decodeDartDefines(environment.defines, kDartDefines))
+          '-D$dartDefine',
+      ];
+    }
+
+    List<String> setOutputFile(String output) {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart:175]: I don't think this one is really carrying its weight

**Review by jonahwilliams — APPROVED** [review]
LGTM with nits

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 3]
```diff
@@ -153,64 +153,49 @@ class Dart2JSTarget extends Target {
...
-    const String packageFile = '.packages';
-    final File outputKernel = environment.buildDir.childFile('app.dill');
-    final File outputFile = environment.buildDir.childFile('main.dart.js');
-    final List<String> dartDefines = decodeDartDefines(environment.defines, kDartDefines);
-    final List<String> extraFrontEndOptions = decodeDartDefines(environment.defines, kExtraFrontEndOptions);
 
-    // Run the dart2js compilation in two stages, so that icon tree shaking can
-    // parse the kernel file for web builds.
-    final ProcessResult kernelResult = await globals.processManager.run(<String>[
+    final List<String> initCommand = <String>[
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart:158]: since this isn't a function any more I would name it something like "sharedArgs"

Comment (includecmath) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart]: How about sharedCommandOptions?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart]: Sounds good!

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 4]
```diff
@@ -153,64 +153,49 @@ class Dart2JSTarget extends Target {
...
-      outputKernel.path,
-      '--packages=$packageFile',
+      '--libraries-spec=${globals.fs.path.join(globals.artifacts.getArtifactPath(Artifact.flutterWebSdk), 'libraries.json')}',
+      ...?decodeDartDefines(environment.defines, kExtraFrontEndOptions),
       if (buildMode == BuildMode.profile)
         '-Ddart.vm.profile=true'
       else
         '-Ddart.vm.product=true',
-      for (final String dartDefine in dartDefines)
+      for (final String dartDefine
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart:168]: could you leave this on the same line?

**Comment by includecmath** [comment]
@jonahwilliams Hi Williams Can we add code format check to above checks? Using unified code style tools show diffs to developers. 
Also if the PR checklist - [ ] change to - [] allow developers to directly check without deleting the space.

**Comment by jonahwilliams** [comment]
>  Hi Williams Can we add code format check to above checks? 

I wish we could, but as a policy we are required to manually format the code

**Review by jonahwilliams — APPROVED** [review]
LGTM


Will land when the tree is green. Thanks for the cleanup!

**Pull request merged**